be/src/storage/rowset/scalar_column_iterator.cpp

```diff
@@ -61,21 +63,62 @@ Status ScalarColumnIterator::init(const ColumnIteratorOptions& opts) {
     if (_reader->encoding_info()->encoding() != DICT_ENCODING) {
         return Status::OK();
     }
-
-    if (_reader->column_type() == TYPE_CHAR) {
+    LogicalType column_type = delegate_type(_reader->column_type());
+    DCHECK(supports_dict_encoding(column_type))
+            << strings::Substitute("dict encoding with unsupported $0 field type", column_type);
+    switch (column_type) {
+    case TYPE_CHAR:
         _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_CHAR>;
-    } else if (_reader->column_type() == TYPE_VARCHAR) {
+        break;
+    case TYPE_VARCHAR:
         _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_VARCHAR>;
-    } else {
-        return Status::NotSupported("dict encoding with unsupported field type");
+        break;
+    case TYPE_SMALLINT:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_SMALLINT>;
+        break;
+    case TYPE_INT:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_INT>;
+        break;
+    case TYPE_BIGINT:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_BIGINT>;
+        break;
+    case TYPE_LARGEINT:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_LARGEINT>;
+        break;
+    case TYPE_FLOAT:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_FLOAT>;
+        break;
+    case TYPE_DOUBLE:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_DOUBLE>;
+        break;
+    case TYPE_DATE:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_DATE>;
+        break;
+    case TYPE_DATETIME:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_DATETIME>;
+        break;
+    case TYPE_DECIMALV2:
+        _init_dict_decoder_func = &ScalarColumnIterator::_do_init_dict_decoder<TYPE_DECIMALV2>;
+        break;
```

Comment:
Does it support DecimalV3?

Reply:
Yes. it supports DECIMALV3. In fact, DECIMALV3(TYPE_DECIMAL32/TYPE_DECIMAL64/TYPE_DECIMAL128) are reprensted by int32_t/int64_t/int128_t